src/hb-ot-cff2-table.hh

```diff
@@ -0,0 +1,386 @@
+          return false;
+        break;
+      case OpCode_StdVW:
+        if (unlikely (!stack.check_pop_real (val.stdVW)))
+          return false;
+        break;
+      case OpCode_BlueScale:
+        if (unlikely (!stack.check_pop_real (val.blueScale)))
+          return false;
+        break;
+      case OpCode_BlueShift:
+        if (unlikely (!stack.check_pop_real (val.blueShift)))
+          return false;
+        break;
+      case OpCode_BlueFuzz:
+        if (unlikely (!stack.check_pop_real (val.blueFuzz)))
+          return false;
+        break;
+      case OpCode_StemSnapH:
+        if (unlikely (!stack.check_pop_delta (val.stemSnapH)))
+          return false;
+        break;
+      case OpCode_StemSnapV:
+        if (unlikely (!stack.check_pop_delta (val.stemSnapV)))
+          return false;
+        break;
+      case OpCode_LanguageGroup:
+        if (unlikely (!stack.check_pop_int (val.languageGroup)))
+          return false;
+        break;
+      case OpCode_ExpansionFactor:
+        if (unlikely (!stack.check_pop_real (val.expansionFactor)))
+          return false;
+        break;
+      case OpCode_Subrs:
+        if (unlikely (!check_pop_offset (stack, val.subrsOffset)))
+          return false;
+        break;
+      case OpCode_blend:
+        // XXX: TODO
+        break;
+
+      default:
+        return false;
+    }
+    return true;
+  }
+};
+
+typedef Interpreter<CFF2TopDictOpSet, CFF2TopDictValues> CFF2TopDict_Interpreter;
+typedef Interpreter<CFF2FontDictOpSet, CFF2FontDictValues> CFF2FontDict_Interpreter;
+typedef Interpreter<CFF2PrivateDictOpSet, CFF2PrivateDictValues> CFF2PrivateDict_Interpreter;
+
+}; /* namespace CFF */
+
+namespace OT {
+
+using namespace CFF;
+
+struct cff2
+{
+  static const hb_tag_t tableTag        = HB_OT_TAG_cff2;
+
+  inline bool sanitize (hb_sanitize_context_t *c) const
+  {
+    TRACE_SANITIZE (this);
+    return_trace (c->check_struct (this) &&
+                  version.sanitize (c) &&
+                  likely (version.major == 2));
+  }
+
+  struct accelerator_t
+  {
+    inline void init (hb_face_t *face)
+    {
+      this->blob = OT::Sanitizer<OT::cff2>().sanitize (face->reference_table (HB_OT_TAG_cff2));
+      const OT::cff2 *cff2 = this->blob->as<OT::cff2> ();
+
+      if (cff2 == &Null(OT::cff2))
+      {
+        hb_blob_destroy (blob);
+        return;
+      }
+
+      { /* parse top dict */
+        ByteStr topDictStr (cff2 + cff2->topDict, cff2->topDictSize);
+        CFF2TopDict_Interpreter top_interp;
+        top_interp.init ();
+        top_interp.interpret (topDictStr, top);
+        top_interp.fini ();
+      }
+      
+      varStore = &OT::StructAtOffset<VariationStore>(cff2, top.vstoreOffset);
+      charStrings = &OT::StructAtOffset<CharStrings>(cff2, top.charStringsOffset);
+      fdArray = &OT::StructAtOffset<FDArray>(cff2, top.FDArrayOffset);
+      fdSelect = &OT::StructAtOffset<FDSelect>(cff2, top.FDSelectOffset);
```

Comment:
As the comment below mentions you need to sanitize these to ensure it's safe to access the memory they point too.

Reply:
Added a sanitize call to each if not Null